plugins/core/src/terminal-service.ts

```diff
@@ -0,0 +1,73 @@
+import { ScryptedDeviceBase, StreamService } from "@scrypted/sdk";
+import type { IPty, spawn as ptySpawn } from 'node-pty-prebuilt-multiarch';
+import { createAsyncQueue } from '@scrypted/common/src/async-queue'
+
+export const TerminalServiceNativeId = 'terminalservice';
+
+export class TerminalService extends ScryptedDeviceBase implements StreamService {
+    async connectStream(input: AsyncGenerator<any, void>): Promise<AsyncGenerator<any, void>> {
+        const spawn = require('node-pty-prebuilt-multiarch').spawn as typeof ptySpawn;
+        const cp: IPty = spawn(process.env.SHELL, [], {});
+        const queue = createAsyncQueue<Buffer>();
+        cp.onExit(() => queue.end());
```

Comment:
does cp have an error event? may need to listen and call queue.end(error) in that case?

Reply:
It doesn't look like it, the typing here: https://github.com/microsoft/node-pty/blob/main/typings/node-pty.d.ts#L108